ethernet_interface.cpp

```diff
@@ -282,6 +283,10 @@ std::string
     static_assert(sizeof(ifr.ifr_hwaddr.sa_data) >= sizeof(ether_addr));
     std::string_view hwaddr(reinterpret_cast<char*>(ifr.ifr_hwaddr.sa_data),
                             sizeof(ifr.ifr_hwaddr.sa_data));
+    if (sock)
```

Comment:
sock needs to be also closed at line 280 - before returning

Reply:
Done